### PR TITLE
Add improvements to terraform_docs.sh

### PR DESCRIPTION
**Pull request by jburnham**
Support adding hooks to README.md files that didn't have the hook before.

Remove the additional newline getting added to the docs causing the pre-commit
to fail twice on first getting the file added to a project.

### CODE REVIEW

File: terraform_docs.sh [hunk 1]
```diff
@@ -23,11 +23,22 @@ for path_uniq in $(echo "${paths[*]}" | tr ' ' '\n' | sort -u); do
...
+      terraform-docs md ./ | awk 'NR>1{print buf}{buf = $0}' >> README.md
+      echo "<!-- END OF TERRAFORM-DOCS HOOK -->" >> README.md
+      echo "Updating README.md, please git add."
+      exit 1
+    fi
   else
     echo -n "<!-- BEGINNING OF TERRAFORM-DOCS HOOK -->" > README.md
-    terraform-docs md ./ >> README.md
+    # Awk is used to remove the final newline
+    terraform-docs md ./ | awk 'NR>1{print buf}{buf = $0}' >> README.md
```
Comment (mrwacky42) [inline terraform_docs.sh]: Shirley this is better served by `sed`.

**Review by mrwacky42 — DISMISSED** [review]

File: terraform_docs.sh (hunk 1 above)
Comment (mrwacky42) [reply terraform_docs.sh]: or this "nice tidy" Python solution - https://stackoverflow.com/a/1663283

Comment (mrwacky42) [reply terraform_docs.sh]: or.. MOAR Perl:
`... | perl -p -e 'chomp if eof' > README.md`

**Review by rccolaaa — DISMISSED** [review]

File: terraform_docs.sh (hunk 1 above)
Comment (mrwacky42) [reply terraform_docs.sh]: Too much carrot cake, and assuming single byte characters, and assuming a trailing newline:
```
terraform-docs md ./ > README.tmp
COUNT=$(cat README.tmp |wc -c)
dd if=README.tmp of=README.md bs=1 count=$((COUNT-1))
rm README.tmp
```

**Review by mrwacky42 — APPROVED** [review]

**Review by rccolaaa — APPROVED** [review]
*both* of em seem unintuitive, but yeah `sed` is at least cleaner

**Pull request merged**